src/main/java/org/springframework/data/elasticsearch/repository/ElasticsearchRepository.java

```diff
@@ -55,23 +56,32 @@
 	<S extends T> S indexWithoutRefresh(S entity);
 
 	/**
-	 * @deprecated since 4.0, use standard repository method naming or @{@link Query} annotated methods, or
-	 *             {@link org.springframework.data.elasticsearch.core.ElasticsearchOperations}.
+	 * @deprecated since 4.0, use {@link #searchQuery(Query)}, standard repository method naming or @{@link Query}
+	 *             annotated methods, or {@link org.springframework.data.elasticsearch.core.ElasticsearchOperations}.
 	 */
+	@Deprecated
 	Iterable<T> search(QueryBuilder query);
 
 	/**
-	 * @deprecated since 4.0, use standard repository method naming or @{@link Query} annotated methods, or
-	 *             {@link org.springframework.data.elasticsearch.core.ElasticsearchOperations}.
+	 * @deprecated since 4.0, use {@link #searchQuery(Query)}, standard repository method naming or @{@link Query}
+	 *             annotated methods, or {@link org.springframework.data.elasticsearch.core.ElasticsearchOperations}.
 	 */
+	@Deprecated
 	Page<T> search(QueryBuilder query, Pageable pageable);
 
 	/**
-	 * @deprecated since 4.0, use standard repository method naming or @{@link Query} annotated methods, or
-	 *             {@link org.springframework.data.elasticsearch.core.ElasticsearchOperations}.
+	 * @deprecated since 4.0, use {@link #searchQuery(Query)}, standard repository method naming or @{@link Query}
+	 *             annotated methods, or {@link org.springframework.data.elasticsearch.core.ElasticsearchOperations}.
 	 */
 	Page<T> search(Query searchQuery);
 
+	/**
+	 * execute the given query and return the result in a SearchPage.
+	 * 
+	 * @since 4.1
+	 */
+	SearchPage<T> searchQuery(Query query);
```

Comment:
We should get actually rid of these methods entirely as that's an anti-pattern in Spring Data. See also https://jira.spring.io/browse/DATAMONGO-2610.